operator/watchers/service_cache.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package watchers
+
+import (
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+func newServiceCacheCallback(swgSvcs, swgEps *lock.StoppableWaitGroup) *serviceCacheCallback {
+	return &serviceCacheCallback{
+		swgSvcs: swgSvcs,
+		swgEps:  swgEps,
+	}
+}
+
+// serviceCacheCallback represents an object containing K8s event reaction
+// logic for the K8sSvcCache. It implements callbacks.ServiceHandler and is
+// used in the watcher.
+type serviceCacheCallback struct {
+	swgSvcs, swgEps *lock.StoppableWaitGroup
+}
+
+func (c *serviceCacheCallback) OnAdd(obj *slim_corev1.Service) {
+	log.WithField(logfields.ServiceName, obj.Name).Debugf("Received service addition %+v", obj)
```

Comment:
I'm not sure if it is worth duping the entire object into debug messages. I know it must have been useful while developing but it doesn't seem to be useful now the functionality is implemented. Consider duping only key-fields that matter to debug an issue.

Reply:
I copied these log statements straight from the service watcher callbacks, so we were already doing this. I was just trying to keep the code the same.